src/components/code-editor/index.js

```diff
@@ -4,43 +4,45 @@ module.exports = {
         return {
             showSpinner: false,
 	        expectingSave: false,
-	        saving: false
+	        saving: false,
+	        currentFile: null
         }
     },
-    props: ['context', 'file', 'parent'],
+    props: ['context', 'file', 'messages'],
     created: function() {
+        this.currentFile = this.file;
         this.startListener();
     },
     methods: {
 	startListener: function() {
+	    var that = this;
 	    var iframe = document.getElementById("editor");
 	    if (iframe == null) {
-    		setTimeout(this.startListener, 500);
+    		setTimeout(function(){that.startListener();}, 1000);
```

Comment:
do we need the function wrapper?

Reply:
yes we do. I have been caught out by that before. For fun test it out yourself. Without the function it will call print() immediately
 	setTimeout(print("hello"), 10000);
      function print(msg) {
      	console.log("msg:" + msg);
      }

Reply:
I disagree. print() is a function call, print is a function pointer.

Reply:
ahh - yes

Reply:
reverted back to original